### PR TITLE
Migrate Kubernetes extension commands #3990

### PR DESCRIPTION
**Pull request by vrubezhny**
Fixes: #3990

### CODE REVIEW

**Comment by vrubezhny** [comment]
Loading Deployment/Pod YAMLs while the k8s extension isn't installed:

[Screencast from 2024-05-08 02-43-03.webm](https://github.com/redhat-developer/vscode-openshift-tools/assets/620781/0db86f3e-9fe6-47f1-9361-dc9611ae8cc6)

**Comment by vrubezhny** [comment]
The 'release/yml' is to be updated as we don't need the `ms-kubernetes-tools.vscode-kubernetes-tools` extension dependency anymore:

https://github.com/redhat-developer/vscode-openshift-tools/blob/49d16c5d48512d62706e12ef0f1d8f69055dbe5f/.github/workflows/release.yml#L52

**Comment by vrubezhny** [comment]
> The 'release.yml' is to be updated as we don't need the `ms-kubernetes-tools.vscode-kubernetes-tools` extension dependency anymore:
> 
> https://github.com/redhat-developer/vscode-openshift-tools/blob/49d16c5d48512d62706e12ef0f1d8f69055dbe5f/.github/workflows/release.yml#L52

@rgrunber Could you please review the change made to `release.yml` (the k8s extension is not added to the `extensionDependencies`  anymore during release)?

**Comment by rgrunber** [comment]
The change there looks fine. I think this could have been removed a while ago since it was already present in the package.json. Maybe we kept it because integration tests would fail if it was in the package.json, but  that issue was resolved : https://github.com/redhat-developer/vscode-openshift-tools/issues/3091 .

File: src/k8s/vfs/kuberesources.virtualfs.ts
Comment (datho7561) [inline src/k8s/vfs/kuberesources.virtualfs.ts:113]: It seems like the virtual filesystem has support for loading installed helm charts as a yaml document, but currently we don't allow opening installed helm charts from the tree view. I think it might be a good idea to allow loading helm installations.

I think it's your call whether we do this in this PR or in a future one.

**Review by datho7561 — APPROVED** [review]
Looks good and works well. See the comment about Helm support.

File: src/k8s/vfs/kuberesources.virtualfs.ts
Comment (vrubezhny) [reply src/k8s/vfs/kuberesources.virtualfs.ts:113]: I'll probably file a new issue for this.

Another thing is `vs-kubernetes.vs-kubernetes.outputFormat` preference specifying the file format for loading a deployment config (`json` or `yaml`): with this PR we're getting rid of k8s extension dependency but still trying to read its preference value:

 https://github.com/redhat-developer/vscode-openshift-tools/blob/ef004c7c08af73d2dae90a02c6082e3eb313098f/src/explorer.ts#L555C2-L565C6

I believe we need to define and manage this preference inside `vscode-openshift-tools`

**Comment by vrubezhny** [comment]
Added `openshiftToolkit.outputFormat` preference in order to control the format of output for deployment documents. So the Deployment/Pod configuration documents can be opened as `yaml` (default) or  `json` documents.

**Comment by datho7561** [comment]
Doesn't seem to compile

**Comment by vrubezhny** [comment]
Loading of Helm Chart Configuration document is added:

[Screencast from 2024-05-09 15-23-09.webm](https://github.com/redhat-developer/vscode-openshift-tools/assets/620781/3a29a980-e695-43e3-a857-c3d47347dafe)

**Comment by vrubezhny** [comment]
@datho7561 Could you please take a look at the latest additions: 
- Preference added for selection of output format - https://github.com/redhat-developer/vscode-openshift-tools/pull/4106#issuecomment-2101216476
- Load Helm Chart configuration document action - https://github.com/redhat-developer/vscode-openshift-tools/pull/4106#issuecomment-2103228259

**Review by datho7561 — APPROVED** [review]
Other than the comment about double-click opening not working for Helm installations (which I think is not a bit deal), this PR seems to be great! Let me know if you want to fix that before merging or merge as-is.

**Comment by vrubezhny** [comment]
> Other than the comment about double-click opening not working for Helm installations (which I think is not a bit deal), this PR seems to be great! Let me know if you want to fix that before merging or merge as-is.

Forgot about this. I'll  fix it

**Review by datho7561 — APPROVED** [review]
Looks good. The Serverless failure is legitimate, but unrelated.

**Pull request merged**
